src/components/KTable/ColumnVisibilityMenu.vue

```diff
@@ -0,0 +1,141 @@
+<template>
+  <div class="table-column-visibility-menu">
+    <KDropdown
+      data-testid="table-column-visibility-menu"
+      @toggle-dropdown="handleDropdownToggle"
+    >
+      <KTooltip text="Show/Hide Columns">
+        <KButton
+          appearance="secondary"
+          class="menu-button"
```

Comment:
issue: since this button will essentially be rendered in KTable toolbar - where often times we will have KInput and/or KSelect, this button needs to have `size="large"` to render equal size next to those components. However it will be too big when it's the only thing in the toolbar which isn't ideal - so maybe this is something we can control with a prop? Something like `columnVisibilityButtonSize` - seems oddly specific and I know @adamdehaven will hate it (so do I) but I'm sure displaying medium size button next to KInputs will be ab issue down the line

Reply:
If we have inputs in the toolbar regularly, then just make the button size large by default? It shouldn't "change" depending on the siblings; we want consistency

Reply:
^ Works for me

Reply:
![image](https://github.com/Kong/kongponents/assets/67973710/c4d68810-d67f-4193-b6c4-130399f5b6f1)

![image](https://github.com/Kong/kongponents/assets/67973710/7002ad2b-ac89-4bbf-ad3c-cbdc63a7e230)

Reply:
(regardless of the button/icon size) IMO the button just kind of "floating" there by itself is a bit odd; e.g. there's nothing anchoring it to the table or showing that it's related